worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/shovel/ShovelDistributor.java

```diff
@@ -107,42 +108,39 @@ public void runOnce() {
             }
             else {
                 for(final Map.Entry<Queue, Long> queueConsumptionTarget: consumptionTarget.entrySet()) {
-//                    if(sourceQueue.getMessages() == 0) {
-//                        LOGGER.info("Source queue '{}' has no messages, ignoring.", 
-//                                distributorWorkItem.getTargetQueue().getName());
-//
-//                        continue;
-//                    }
-                    if(retrievedShovels.stream().anyMatch(s -> s.getName()
-                            .endsWith(queueConsumptionTarget.getKey().getName()))) {
-                        LOGGER.info("Shovel {} already exists, ignoring.", queueConsumptionTarget.getKey().getName());
+
+                    final String shovelName = queueConsumptionTarget.getKey().getName();
+
+                    final boolean nonTerminatedShovelExists = retrievedShovels.stream()
+                            .filter(retrievedShovel -> retrievedShovel.getState() != ShovelState.TERMINATED)
```

Comment:
Small point but you could combine the check for state and name endsWith into anyMatch.